src/evaluator.rs

```diff
@@ -177,7 +177,11 @@ impl<'src, 'run> Evaluator<'src, 'run> {
       }
       Expression::Group { contents } => self.evaluate_expression(contents),
       Expression::Join { lhs, rhs } => {
-        Ok(self.evaluate_expression(lhs)? + "/" + &self.evaluate_expression(rhs)?)
+        let lhs = match lhs {
```

Comment:
How about:

```
Expression::Join { None, rhs } => {
  Ok("/" + self.evaluate_expression(lhs)?) // not sure about the + here, might need String::from("/") + or something else
}
Expression::Join { Some(lhs), rhs } => {
  Ok(lhs + "/" + &self.evaluate_expression(rhs)?)
}
```

In general, I think it's cleaner to just handle the cases separately, rather than creating an empty string to join with.

Reply:
Ah, so, I had tried this earlier but it wasn't working. Taking a look at it again I see that the enum can be used in the match statement like this: `Expression::Join { lhs: None, rhs } => ()`